.travis.yml

```diff
@@ -11,8 +11,8 @@ before_install:
   - git clone --depth=50 --branch=master https://github.com/jupyterlab/jupyterlab.git
   - cd jupyterlab
 install:
-  - npm install fs fs-extra child-process path sort-package-json glob
-  - npm run addsibling $TRAVIS_BUILD_DIR && npm install
+  - npm install --no-optional
+  - npm run addsibling $TRAVIS_BUILD_DIR && npm install --no-optional
```

Comment:
Cool @ian-r-rose. I didn't know that --no-optional was available. Quick question for my own understanding. Why does `npm install` execute again on line 15?

Reply:
This CI setup is kind of tricky, in that it builds against JupyterLab master, rather than the released sources. In order to do this, we need to graft it into the JupyterLab source tree as a full sibling package using the `addsibling` script.

The first `npm install` installs the dependencies for JupyterLab, but it also installs the things needed to make the `addsibling` script work. The second `npm install` installs the dependencies for the new `@jupyterlab/google-drive` sibling package.

The `--no-optional` command is mostly for cosmetic purposes, as there is a node error on rebuilding some node canvas package that we don't need.

Reply:
Thanks for the detailed explanation. Makes much more sense now.

Reply:
You might consider adding your above explanation to the contributor/dev docs too.

Reply:
@jzf2101 fyi

Reply:
Yeah, I think we want to clean this approach up and document it for general extensions that want to test against master (as I assume people will want to do).